### PR TITLE
Revise Pluggable resolvers

### PR DESCRIPTION
**Pull request by desandro**
@sheerun Can you look over example code? I find it helpful, but not sure if code is correct :P

+ Use page template (not homepage)
+ add JS example code
+ simplify parameters & returns
+ `Type` style

### CODE REVIEW

File: docs/pluggable-resolvers.md [hunk 1]
```diff
@@ -122,89 +119,98 @@ module.exports = function resolver (bower) {
...
 Downloads given endpoint and returns path to temporary directory.
 
-*Parameters:*
-
-  * `endpoint: object` - endpoint for the resource to download
-    * `name: string` - name of resource (like `jquery`)
-    * `source: string` - where to download resource from (like `git://github.com/jquery/jquery.git`)
-    * `target: string` - the version or release of resource to download (like `v1.0.0`)
+{% highlight javascript %}
+var fetched = resolver.fetch( endPoint, cached )
```
Comment (sheerun) [inline docs/pluggable-resolvers.md]: canonically "endpoint" :)

**Comment by sheerun** [comment]
Not really. You'll never call resolver's methods directly. It all happens under the hood.

I'd merge it anyway, because I find this style more readable :)

**Comment by paulohp** [comment]
@desandro @sheerun This's still valid? Can I review and merge? 

Thanks

**Comment by desandro** [comment]
Yes, should be good to merge. Although, the technical portions need an overview. I think we need to provide clear examples with code on how Pluggable Resolvers are used.

**Comment by sheerun** [comment]
@paulohp There was another review merged before this one, so expect lot of conflicts merging this one..

**Comment by paulohp** [comment]
#157 right ? I'm on the way to create some code examples here. Can I commit on this PR @desandro @sheerun ?

**Comment by sheerun** [comment]
Sure :)

**Comment by paulohp** [comment]
@desandro @sheerun I did the fix for the conflicts and added a link to a external code example. Can you guys review?

**Comment by desandro** [comment]
Reviewed. :+1:

File: docs/pluggable-resolvers.md [hunk 2]
```diff
@@ -128,95 +118,102 @@ module.exports = function resolver (bower) {
...
-*Parameters:*
-
-  * `source: string` - source from bower.json, like `git://github.com/jquery/jquery.git`
+{% highlight javascript %}
+var isMatched = resolver.match( source )
+{% endhighlight %}
 
-*Returns* a boolean that tells whether resolver can handle given source (either by locating them with `locate` method, or fetching it with `fetch` + optional `releases` method).
+  * `source`: `String` - source from bower.json, like `git://github.com/jquery/jquery.git`
+  * `isMatched`: `Boolean` - *Returns* a boolean that tells whether resolver can handle given source (either by locating them with `locate` method, or fetching it with `fetch` + optional `releases` method).
```
Comment (sheerun) [inline docs/pluggable-resolvers.md]: Without "*returns* a" should be better. The same in other places :)

**Comment by sheerun** [comment]
Looks ok. You can merge :)

**Comment by faceleg** [comment]
:tada:

**Pull request merged**